wire-library/wire-grpc-java-server/src/main/java/com/squareup/wire/kotlin/grpcserver/AddBlockingStub.kt

```diff
@@ -0,0 +1,61 @@
+package com.squareup.wire.kotlin.grpcserver
+
+import com.squareup.kotlinpoet.ClassName
+import com.squareup.kotlinpoet.CodeBlock
+import com.squareup.kotlinpoet.FunSpec
+import com.squareup.kotlinpoet.MemberName
+import com.squareup.kotlinpoet.ParameterizedTypeName.Companion.parameterizedBy
+import com.squareup.kotlinpoet.TypeSpec
+import com.squareup.kotlinpoet.asClassName
+import com.squareup.wire.schema.Rpc
+import com.squareup.wire.schema.Service
+import io.grpc.Channel
+import io.grpc.stub.ClientCalls
+
+internal fun TypeSpec.Builder.addBlockingStub(service: Service): TypeSpec.Builder = this
+  .addFunction(
+    FunSpec.builder("newBlockingStub")
+      .addParameter("channel", Channel::class)
+      .returns(ClassName("", "${service.name}BlockingStub"))
+      .addCode("return %T(channel)", ClassName("", "${service.name}BlockingStub"))
+      .build()
+  )
+  .addType(
+    TypeSpec.classBuilder("${service.name}BlockingStub")
+      .addAbstractStubConstructor(service)
+      .addBlockingStubRpcCalls(service)
+      .build()
+  )
+
+private fun TypeSpec.Builder.addBlockingStubRpcCalls(service: Service): TypeSpec.Builder {
```

Comment:
These seem like strange uses of extension functions. Extension functions are meant for adding reusable methods to existing classes, whereas it seems here you are using it as a way to enable method chaining. It makes the implementation a bit hard to follow. I think you should be creating methods on your KotlinGrpcGenerator class that take `TypeSpec.Builder` as an argument instead, or separate "Generator" classes for each component if one is too big

Reply:
That's a fair point about using extension functions here. 

Each of the parts of the generated code are a bit big and I'd like to test them separately. I don't think we should use a class to house a single method though, kotlin allows for top-level functions after all.

Reply:
You could make several classes, each responsible for one of the components that get generated? I'd just follow the patterns in https://github.com/square/wire/blob/master/wire-library/wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt